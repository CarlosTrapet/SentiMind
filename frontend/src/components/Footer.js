import React from 'react';
import reactLogo from '../images/react-original.svg';
import pythonLogo from '../images/python-original.svg';
import dockerLogo from '../images/docker-original.svg';
import amazonLogo from '../images/amazonwebservices-original.svg';
import githubLogo from '../images/github-original.svg';

const Footer = () => {
  return (
    <footer id="footer">
      <p id="sentibot-info">
        <em>
          * SentiMind analyses and visualises the sentiment (frequency and
          polarity) of the latest 100 tweets using SentiBot, an algorithm
          trained using machine learning
        </em>
      </p>

      <p id="tech-stack">
        Powered by{' '}
        <a target="_blank" href="https://reactjs.org/">
          <img
            className="logos"
            alt="react logo"
            width="40"
            height="40"
            src={reactLogo}
          />
        </a>{' '}
        and{' '}
        <a target="_blank" href="https://www.python.org/">
          <img
            className="logos"
            alt="python logo"
            width="40"
            height="40"
            src={pythonLogo}
          />
        </a>{' '}
        deployed with{' '}
        <a target="_blank" href="https://www.docker.com/">
          <img
            className="logos"
            alt="python logo"
            width="40"
            height="40"
            src={dockerLogo}
          />
        </a>{' '}
        and{' '}
        <a target="_blank" href="https://aws.amazon.com/">
          <img
            className="logos"
            alt="python logo"
            width="40"
            height="40"
            src={amazonLogo}
          />
        </a>{' '}
        check our{' '}
        <a target="_blank" href="https://github.com/marcusfgardiner/SentiMind">
          <img
            className="logos"
            alt="python logo"
            width="40"
            height="40"
            src={githubLogo}
          />
        </a>{' '}
        for more info.
      </p>
    </footer>
  );
};

export default Footer;
