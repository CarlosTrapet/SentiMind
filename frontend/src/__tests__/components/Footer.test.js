import React from 'react';
import { shallow } from 'enzyme';
import Footer from '../../components/Footer';

describe('Footer', () => {
  const footer = shallow(<Footer />);

  it('renders correctly', () => {
    expect(footer).toMatchSnapshot();
  });
});
