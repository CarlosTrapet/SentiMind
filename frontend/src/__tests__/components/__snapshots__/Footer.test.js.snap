// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          id="sentibot-info"
>
          <em>
                    * SentiMind analyses and visualises the sentiment (frequency and polarity) of the latest 100 tweets using SentiBot, an algorithm trained using machine learning
          </em>
</p>,
        <p
          id="tech-stack"
>
          Powered by
           
          <a
                    href="https://reactjs.org/"
                    target="_blank"
          >
                    <img
                              alt="react logo"
                              className="logos"
                              height="40"
                              src="react-original.svg"
                              width="40"
                    />
          </a>
           
          and
           
          <a
                    href="https://www.python.org/"
                    target="_blank"
          >
                    <img
                              alt="python logo"
                              className="logos"
                              height="40"
                              src="python-original.svg"
                              width="40"
                    />
          </a>
           
          Deployed with
           
          <a
                    href="https://www.docker.com/"
                    target="_blank"
          >
                    <img
                              alt="python logo"
                              className="logos"
                              height="40"
                              src="docker-original.svg"
                              width="40"
                    />
          </a>
           
          and
           
          <a
                    href="https://aws.amazon.com/"
                    target="_blank"
          >
                    <img
                              alt="python logo"
                              className="logos"
                              height="40"
                              src="amazonwebservices-original.svg"
                              width="40"
                    />
          </a>
           
          Check our
           
          <a
                    href="https://github.com/marcusfgardiner/SentiMind"
                    target="_blank"
          >
                    <img
                              alt="python logo"
                              className="logos"
                              height="40"
                              src="github-original.svg"
                              width="40"
                    />
          </a>
           
          for more info.
</p>,
      ],
      "id": "footer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <em>
            * SentiMind analyses and visualises the sentiment (frequency and polarity) of the latest 100 tweets using SentiBot, an algorithm trained using machine learning
</em>,
          "id": "sentibot-info",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "* SentiMind analyses and visualises the sentiment (frequency and polarity) of the latest 100 tweets using SentiBot, an algorithm trained using machine learning",
          },
          "ref": null,
          "rendered": "* SentiMind analyses and visualises the sentiment (frequency and polarity) of the latest 100 tweets using SentiBot, an algorithm trained using machine learning",
          "type": "em",
        },
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Powered by",
            " ",
            <a
              href="https://reactjs.org/"
              target="_blank"
>
              <img
                            alt="react logo"
                            className="logos"
                            height="40"
                            src="react-original.svg"
                            width="40"
              />
</a>,
            " ",
            "and",
            " ",
            <a
              href="https://www.python.org/"
              target="_blank"
>
              <img
                            alt="python logo"
                            className="logos"
                            height="40"
                            src="python-original.svg"
                            width="40"
              />
</a>,
            " ",
            "Deployed with",
            " ",
            <a
              href="https://www.docker.com/"
              target="_blank"
>
              <img
                            alt="python logo"
                            className="logos"
                            height="40"
                            src="docker-original.svg"
                            width="40"
              />
</a>,
            " ",
            "and",
            " ",
            <a
              href="https://aws.amazon.com/"
              target="_blank"
>
              <img
                            alt="python logo"
                            className="logos"
                            height="40"
                            src="amazonwebservices-original.svg"
                            width="40"
              />
</a>,
            " ",
            "Check our",
            " ",
            <a
              href="https://github.com/marcusfgardiner/SentiMind"
              target="_blank"
>
              <img
                            alt="python logo"
                            className="logos"
                            height="40"
                            src="github-original.svg"
                            width="40"
              />
</a>,
            " ",
            "for more info.",
          ],
          "id": "tech-stack",
        },
        "ref": null,
        "rendered": Array [
          "Powered by",
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="react logo"
                className="logos"
                height="40"
                src="react-original.svg"
                width="40"
/>,
              "href": "https://reactjs.org/",
              "target": "_blank",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "react logo",
                "className": "logos",
                "height": "40",
                "src": "react-original.svg",
                "width": "40",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "a",
          },
          " ",
          "and",
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="python logo"
                className="logos"
                height="40"
                src="python-original.svg"
                width="40"
/>,
              "href": "https://www.python.org/",
              "target": "_blank",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "python logo",
                "className": "logos",
                "height": "40",
                "src": "python-original.svg",
                "width": "40",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "a",
          },
          " ",
          "Deployed with",
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="python logo"
                className="logos"
                height="40"
                src="docker-original.svg"
                width="40"
/>,
              "href": "https://www.docker.com/",
              "target": "_blank",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "python logo",
                "className": "logos",
                "height": "40",
                "src": "docker-original.svg",
                "width": "40",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "a",
          },
          " ",
          "and",
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="python logo"
                className="logos"
                height="40"
                src="amazonwebservices-original.svg"
                width="40"
/>,
              "href": "https://aws.amazon.com/",
              "target": "_blank",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "python logo",
                "className": "logos",
                "height": "40",
                "src": "amazonwebservices-original.svg",
                "width": "40",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "a",
          },
          " ",
          "Check our",
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="python logo"
                className="logos"
                height="40"
                src="github-original.svg"
                width="40"
/>,
              "href": "https://github.com/marcusfgardiner/SentiMind",
              "target": "_blank",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "python logo",
                "className": "logos",
                "height": "40",
                "src": "github-original.svg",
                "width": "40",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "a",
          },
          " ",
          "for more info.",
        ],
        "type": "p",
      },
    ],
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            id="sentibot-info"
>
            <em>
                        * SentiMind analyses and visualises the sentiment (frequency and polarity) of the latest 100 tweets using SentiBot, an algorithm trained using machine learning
            </em>
</p>,
          <p
            id="tech-stack"
>
            Powered by
             
            <a
                        href="https://reactjs.org/"
                        target="_blank"
            >
                        <img
                                    alt="react logo"
                                    className="logos"
                                    height="40"
                                    src="react-original.svg"
                                    width="40"
                        />
            </a>
             
            and
             
            <a
                        href="https://www.python.org/"
                        target="_blank"
            >
                        <img
                                    alt="python logo"
                                    className="logos"
                                    height="40"
                                    src="python-original.svg"
                                    width="40"
                        />
            </a>
             
            Deployed with
             
            <a
                        href="https://www.docker.com/"
                        target="_blank"
            >
                        <img
                                    alt="python logo"
                                    className="logos"
                                    height="40"
                                    src="docker-original.svg"
                                    width="40"
                        />
            </a>
             
            and
             
            <a
                        href="https://aws.amazon.com/"
                        target="_blank"
            >
                        <img
                                    alt="python logo"
                                    className="logos"
                                    height="40"
                                    src="amazonwebservices-original.svg"
                                    width="40"
                        />
            </a>
             
            Check our
             
            <a
                        href="https://github.com/marcusfgardiner/SentiMind"
                        target="_blank"
            >
                        <img
                                    alt="python logo"
                                    className="logos"
                                    height="40"
                                    src="github-original.svg"
                                    width="40"
                        />
            </a>
             
            for more info.
</p>,
        ],
        "id": "footer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <em>
              * SentiMind analyses and visualises the sentiment (frequency and polarity) of the latest 100 tweets using SentiBot, an algorithm trained using machine learning
</em>,
            "id": "sentibot-info",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "* SentiMind analyses and visualises the sentiment (frequency and polarity) of the latest 100 tweets using SentiBot, an algorithm trained using machine learning",
            },
            "ref": null,
            "rendered": "* SentiMind analyses and visualises the sentiment (frequency and polarity) of the latest 100 tweets using SentiBot, an algorithm trained using machine learning",
            "type": "em",
          },
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Powered by",
              " ",
              <a
                href="https://reactjs.org/"
                target="_blank"
>
                <img
                                alt="react logo"
                                className="logos"
                                height="40"
                                src="react-original.svg"
                                width="40"
                />
</a>,
              " ",
              "and",
              " ",
              <a
                href="https://www.python.org/"
                target="_blank"
>
                <img
                                alt="python logo"
                                className="logos"
                                height="40"
                                src="python-original.svg"
                                width="40"
                />
</a>,
              " ",
              "Deployed with",
              " ",
              <a
                href="https://www.docker.com/"
                target="_blank"
>
                <img
                                alt="python logo"
                                className="logos"
                                height="40"
                                src="docker-original.svg"
                                width="40"
                />
</a>,
              " ",
              "and",
              " ",
              <a
                href="https://aws.amazon.com/"
                target="_blank"
>
                <img
                                alt="python logo"
                                className="logos"
                                height="40"
                                src="amazonwebservices-original.svg"
                                width="40"
                />
</a>,
              " ",
              "Check our",
              " ",
              <a
                href="https://github.com/marcusfgardiner/SentiMind"
                target="_blank"
>
                <img
                                alt="python logo"
                                className="logos"
                                height="40"
                                src="github-original.svg"
                                width="40"
                />
</a>,
              " ",
              "for more info.",
            ],
            "id": "tech-stack",
          },
          "ref": null,
          "rendered": Array [
            "Powered by",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="react logo"
                  className="logos"
                  height="40"
                  src="react-original.svg"
                  width="40"
/>,
                "href": "https://reactjs.org/",
                "target": "_blank",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "react logo",
                  "className": "logos",
                  "height": "40",
                  "src": "react-original.svg",
                  "width": "40",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "a",
            },
            " ",
            "and",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="python logo"
                  className="logos"
                  height="40"
                  src="python-original.svg"
                  width="40"
/>,
                "href": "https://www.python.org/",
                "target": "_blank",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "python logo",
                  "className": "logos",
                  "height": "40",
                  "src": "python-original.svg",
                  "width": "40",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "a",
            },
            " ",
            "Deployed with",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="python logo"
                  className="logos"
                  height="40"
                  src="docker-original.svg"
                  width="40"
/>,
                "href": "https://www.docker.com/",
                "target": "_blank",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "python logo",
                  "className": "logos",
                  "height": "40",
                  "src": "docker-original.svg",
                  "width": "40",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "a",
            },
            " ",
            "and",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="python logo"
                  className="logos"
                  height="40"
                  src="amazonwebservices-original.svg"
                  width="40"
/>,
                "href": "https://aws.amazon.com/",
                "target": "_blank",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "python logo",
                  "className": "logos",
                  "height": "40",
                  "src": "amazonwebservices-original.svg",
                  "width": "40",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "a",
            },
            " ",
            "Check our",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="python logo"
                  className="logos"
                  height="40"
                  src="github-original.svg"
                  width="40"
/>,
                "href": "https://github.com/marcusfgardiner/SentiMind",
                "target": "_blank",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "python logo",
                  "className": "logos",
                  "height": "40",
                  "src": "github-original.svg",
                  "width": "40",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "a",
            },
            " ",
            "for more info.",
          ],
          "type": "p",
        },
      ],
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
